cmd/kubeadm/app/phases/selfhosting/podspec_mutation.go

```diff
@@ -18,38 +18,42 @@ package selfhosting
 
 import (
 	"k8s.io/api/core/v1"
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
 	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
 )
 
 // mutatePodSpec makes a Static Pod-hosted PodSpec suitable for self-hosting
-func mutatePodSpec(name string, podSpec *v1.PodSpec) {
-	mutators := map[string][]func(*v1.PodSpec){
+func mutatePodSpec(cfg *kubeadmapi.MasterConfiguration, name string, podSpec *v1.PodSpec) {
```

Comment:
I'm sorry but I really dislike this. I know it's the only solution right now though so I'm letting this follow as a TODO

We should scope down things and pass a struct only with the minimal required information instead of passing everything. Now it's not clear to the reviewer/reader/coder what of these fields actually are being used...

But TODO; needs lots of other changes in the code as well, so out-of-scope for this PR.